pkg/provisioner/ironic/ironic.go

```diff
@@ -1777,7 +1782,7 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 // deletion operation is completed.
 func (p *ironicProvisioner) Detach() (result provisioner.Result, err error) {
 	// Currently the same behavior as Delete()
-	return p.Delete()
+	return p.Delete(true)
```

Comment:
Detaching must not result in a power off or any other potentially destructive actions.